### PR TITLE
problem finding borders added

### PR DESCRIPTION
**Pull request by pratik-2707**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by pratik-2707** [comment]
I am not getting where this is going wrong ,any help please

**Comment by SansPapyrus683** [comment]
@brebenelmihnea do you think this is a good focus problem?

**Comment by envyaims** [comment]
> @brebenelmihnea do you think this is a good focus problem?

i think its fine

**Comment by SansPapyrus683** [comment]
also for the deployment see how focus problems are supposed to be added, i think this modules has other focus problems that you can look to as an example

**Comment by pratik-2707** [comment]
this is the issue i am facing 
error Problem cses-1732 claims to have an internal solution but doesn't

what exactly internal solution means ?

**Comment by SansPapyrus683** [comment]
> i think this modules has other focus problems that you can look to as an example

File: content/6_Advanced/String_Search.mdx [hunk 1]
```diff
@@ -593,14 +593,64 @@ int main() {
...
+<FocusProblem problem="Finding Borders" />
+
+### Solution - Finding Borders
+
+In this problem, we are tasked with finding all border lengths of a given string. A border of a string is a prefix that is also a suffix of the string but not the whole string. We employ a variation of the Knuth-Morris-Pratt algorithm to efficiently solve this problem.
+
+The approach involves constructing a prefix function \( p[] \), which stores the length of the longest proper border ending at each position in the string. We iterate through the string and update \( p[] \) accordingly, taking care to handle cases where the characters encountered do not match. Finally, we traverse \( p[] \) backwards to find all border lengths, as they represent the lengths of the longest proper borders ending at each position.
+
+### Implementation
+
```
Comment (ryanchou-dev) [inline content/6_Advanced/String_Search.mdx:607]: time complexity?

File: content/6_Advanced/String_Search.mdx [hunk 2]
```diff
@@ -593,14 +593,64 @@ int main() {
...
+	int curr = p[n - 1];
+	vector<int> v;
+	while (curr > 0) {
+		v.push_back(curr);
+		curr = p[curr - 1];
+	}
+	sort(v.begin(), v.end());
+	for (int jj : v) { cout << jj << " "; }
+	return 0;
```
Comment (ryanchou-dev) [inline content/6_Advanced/String_Search.mdx:641]:
```suggestion
```

File: content/6_Advanced/String_Search.mdx [hunk 3]
```diff
@@ -593,14 +593,64 @@ int main() {
...
+In this problem, we are tasked with finding all border lengths of a given string. A border of a string is a prefix that is also a suffix of the string but not the whole string. We employ a variation of the Knuth-Morris-Pratt algorithm to efficiently solve this problem.
+
+The approach involves constructing a prefix function \( p[] \), which stores the length of the longest proper border ending at each position in the string. We iterate through the string and update \( p[] \) accordingly, taking care to handle cases where the characters encountered do not match. Finally, we traverse \( p[] \) backwards to find all border lengths, as they represent the lengths of the longest proper borders ending at each position.
+
+### Implementation
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
```
Comment (ryanchou-dev) [inline content/6_Advanced/String_Search.mdx:611]: can you send ss of this code acing?

File: content/6_Advanced/String_Search.mdx [hunk 4]
```diff
@@ -593,14 +593,64 @@ int main() {
...
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	string s;
+	cin >> s;
+	int n = s.length();
+
+	int p[n] = {0};
```
Comment (ryanchou-dev) [inline content/6_Advanced/String_Search.mdx:620]: use vectors for variable-sized arrays

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**